Algebraic_foundations/include/CGAL/number_utils.h

```diff
@@ -316,6 +316,9 @@ NT approximate_sqrt(const NT& nt, Sqrt sqrt)
 template <typename NT>
 NT approximate_sqrt(const NT& nt)
 {
+  // the initial version of this function was using Algebraic_category
+  // for the dispatch but some ring type (like Gmpz) provides a Sqrt
+  // functor even if not being Field_with_sqrt.
```

Comment:
Maybe the hierarchy of category should be extended to have a `Ring_with_approximate_sqrt`.